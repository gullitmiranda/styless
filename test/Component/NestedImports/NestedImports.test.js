import React from 'react';
import renderer from 'react-test-renderer';
import styled from 'styled-components';
import 'jest-styled-components';

test('#29, nested imports', () => {
    const Div = styled.div`
      @import "tmp.less";
      @import (reference) "reference.less";
      &.light {
        color: @text-color;
        background: @layout-sider-background-light;
      }
    
      &.dark {
        background: @layout-header-background;
        color: @layout-trigger-color;
      }
	`;
    expect(renderer.create(<Div/>).toJSON()).toMatchSnapshot();
});