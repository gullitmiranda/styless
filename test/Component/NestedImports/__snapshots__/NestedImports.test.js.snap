// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#29, nested imports 1`] = `
.c0 .test2 {
  color: test2;
}

.c0 .test {
  color: test;
}

.c0 .tmp {
  color: tmp;
}

.c0.light {
  color: pink;
  background: #222;
}

.c0.dark {
  background: #333;
  color: #444;
}

<div
  className="c0"
/>
`;
